packages/flutter/lib/src/widgets/basic.dart

```diff
@@ -6049,22 +6049,66 @@ class Semantics extends SingleChildRenderObjectWidget {
 /// A widget that merges the semantics of its descendants.
 ///
 /// Causes all the semantics of the subtree rooted at this node to be
-/// merged into one node in the semantics tree. For example, if you
-/// have a widget with a Text node next to a checkbox widget, this
-/// could be used to merge the label from the Text node with the
+/// merged into one node in the semantics tree.
+///
+/// If you have a widget with a Text node next to a checkbox
+/// widget, this could be used to merge the label from the Text node with the
 /// "checked" semantic state of the checkbox into a single node that
 /// had both the label and the checked state. Otherwise, the label
 /// would be presented as a separate feature than the checkbox, and
 /// the user would not be able to be sure that they were related.
 ///
+/// {@tool snippet --template=stateful_widget_scaffold}
+///
+/// To see the end result, make sure to set MaterialApp's showSemanticsDebugger
+/// property to `true`. The property is available for WidgetApp and
+/// CupertinoApp as well.
+///
+/// ```dart
+/// bool _isChecked = true;
+///
+/// @override
+/// Widget build(BuildContext context) {
+///   return Center(
+///     child: MergeSemantics( // comment out this line to see difference
+///       child: Padding(
+///         padding: EdgeInsets.symmetric(horizontal: 20.0),
+///         child: Row(
+///           children: <Widget>[
+///             Expanded(child: Text('Checkbox is here')),
+///             Checkbox(
+///               value: _isChecked,
+///               onChanged: (bool newValue) {
+///                 setState(() {
+///                   _isChecked = newValue;
+///                 });
+///               },
+///             ),
+///           ],
+///         ),
+///       ),
+///     ),  // comment out this line to see difference
+///   );
+/// }
+/// ```
+/// {@end-tool}
+///
+/// ## Conflicting Semantics
+///
 /// Be aware that if two nodes in the subtree have conflicting
-/// semantics, the result may be nonsensical. For example, a subtree
-/// with a checked checkbox and an unchecked checkbox will be
-/// presented as checked. All the labels will be merged into a single
+/// semantics, the result may be nonsensical.
+///
+/// For example, a subtree with a checked [Checkbox] and an unchecked [Checkbox]
+/// will be presented as checked. All the labels will be merged into a single
 /// string (with newlines separating each label from the other). If
 /// multiple nodes in the merged subtree can handle semantic gestures,
 /// the first one in tree order will be the one to receive the
 /// callbacks.
+///
+/// If a descendant widget of MergeSemantics requires its own [SemanticsNode]
```

Comment:
This is surprising to me. I would expect that `MergeSemantics` overrides a descendant's wish to have its own SemanticsNode. Maybe the bug is that `MergeSemantics` doesn't work properly if somebody requests to own their own semantics node?

Reply:
In https://github.com/flutter/flutter/issues/31437, a `RichText` widget has a gesture recognizer that requires its semantic node to be explicit. However, this conflicts with `CheckboxListTile`'s desire to merge its descendant's node, which results in the assert above being thrown. 

I assumed that the assert is working as intended, since there is a conflict in what should be done and it's up to the user to decide how to handle their semantics. However, if it is a bug, then we can force `MergeSemantics` to override its descendant's desire to own its own node.

Reply:
I think it makes sense, especially as a safe guard against nonsensical or unintentional merge results.

Reply:
@goderbauer What do you think of this?

Reply:
If it merge the children node even if they have their own boundaries. We might cause some inconsistent between accessibility mode and non accessibility mode.
For example, if we have button inside a button,
in non accessibility mode, we will be able the click the inner button
in accessibility mode, the voice over will be skipping the inner button